### PR TITLE
[SPARK-36617][PYTHON] Fix type hints for `approxQuantile` to support multi-column version

### PR DESCRIPTION
**Pull request by carylee**
### What changes were proposed in this pull request?
Update both `DataFrame.approxQuantile` and `DataFrameStatFunctions.approxQuantile` to support overloaded definitions when multiple columns are supplied.

### Why are the changes needed?
The current type hints don't support the multi-column signature, a form that was added in Spark 2.2 (see [the approxQuantile docs](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.DataFrame.approxQuantile.html).) This change was also introduced to pyspark-stubs (https://github.com/zero323/pyspark-stubs/pull/552). @zero323 asked me to open a PR for the upstream change.

### Does this PR introduce _any_ user-facing change?
This change only affects type hints - it brings the `approxQuantile` type hints up to date with the actual code.

### How was this patch tested?
Ran `./dev/lint-python`.

### CODE REVIEW

**Comment by zero323** [comment]
ok to test

**Comment by zero323** [comment]
Thanks for your contribution @carylee. Could you check the [error](https://github.com/apache/spark/pull/33880/checks?check_run_id=3478166282) and follow the instructions there? Thanks!

File: python/pyspark/sql/dataframe.pyi [hunk 1]
```diff
@@ -270,7 +278,9 @@ class DataFrame(PandasMapOpsMixin, PandasConversionMixin):
     def semanticHash(self) -> int: ...
     def inputFiles(self) -> List[str]: ...
     def writeTo(self, table: str) -> DataFrameWriterV2: ...
-    def to_pandas_on_spark(self, index_col: Optional[Union[str, List[str]]] = None) -> PandasOnSparkDataFrame: ...
+    def to_pandas_on_spark(
```
Comment (zero323) [inline python/pyspark/sql/dataframe.pyi:281]: Let's keep unrelated code as-is for now, without reformatting.

**Comment by zero323** [comment]
Could you also add `[PYTHON]` to the title (it should be `[SPARK-36617][PYTHON] ...`)?

**Comment by carylee** [comment]
I'm not sure what the problems is. My branch is in sync with upstream/master and Github actions are enabled (see below):

Screen Shot 2021-08-31 at 4 39 36 PM

**Comment by zero323** [comment]
Could you check your actions tab (https://github.com/carylee/spark/actions) directly (you might see something like described in [this comment](https://github.com/apache/spark/pull/32400#issuecomment-831051189))?

**Comment by zero323** [comment]
ok to test

**Review by zero323 — APPROVED** [review]
LGTM, subject to passing Jenkins tests.

**Comment by zero323** [comment]
retest this please

**Comment by zero323** [comment]
test this please

**Comment by carylee** [comment]
@zero323 It looks like the tests passed? https://github.com/carylee/spark/actions/runs/1191206733

**Review by HyukjinKwon — APPROVED** [review]
LGTM2. I will leave it to @zero323.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
I think it's good to go and merge since GitHub Actions passed. This change shouldn't break any of tests anyway.

**Comment by zero323** [comment]
Thanks everyone!
